### ISSUE TITLE
Implement FromObject for Object

### ISSUE THREAD

**Issue by ModProg** [noib3/nvim-oxi#73]
This allows to get back a "dynamic" object from functions that return <T: FromObject>.

My workaround is to do:
```rs
#[derive(Deref)]
struct Object(oxi::Object);

impl FromObject for Object {
    fn from_object(object: oxi::Object) -> Result<Self, conversion::Error> {
        Ok(Self(object))
    }
}
```

but this would be nice to have in the library, especially as it would be a 4 line implementation

**Comment by ModProg** [comment]
IIRC this worked at some point

**Comment by noib3** [comment]
closed by 5d07ef6c16b7fb57730d1df7633d35524d7f2a92
